corehq/blobs/interface.py

```diff
@@ -44,13 +46,19 @@ def get(self, identifier, bucket=DEFAULT_BUCKET):
         raise NotImplementedError
 
     @abstractmethod
-    def delete(self, identifier=None, bucket=DEFAULT_BUCKET):
+    def delete(self, identifier=NOT_SET, bucket=NOT_SET):
         """Delete a blob
 
-        :param identifier: The identifier of the object to be deleted. The entire
-        bucket will be deleted if this is not specified.
-        :param bucket: Optional bucket name. This must have the same
-        value that was passed to ``put``.
+        Either one or both of `identifier` or `bucket` parameters
+        is required.
+
+        :param identifier: The identifier of the object to be deleted.
+        The entire bucket will be deleted if this is not provided.
+        :param bucket: Bucket name. This must have the same value that
+        was passed to ``put``. Delete will be attempted in the default
+        bucket if an identifier is given but the bucket is not given.
+        Pass this parameter alone as a keyword argument to delete the
+        entire bucket.
```

Comment:
I suppose combining bucket delete with blob delete does make this function interface a bit more complicated. I like the way `db.delete(bucket='blah')` reads, which is why I went with it originally. I think the implementation is now safe from silly mistakes: You can't easily delete a bucket by accident anymore, and it's backward-compatible with existing code, so I probably won't change it.